### PR TITLE
fix: Have enhanceError support more edge case

### PR DESCRIPTION
**Pull request by dprevost-LMI**
- Fix `enhanceError` not supporting edge cases such as `undefined`
- Bring back more test cases from $$ support, like for errorMesssageBe

### CODE REVIEW

File: src/util/stringUtil.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+export const toJsonString = (value: unknown): string => {
+    if (isString(value)) {
+        return value
+    }
+    try {
+        return JSON.stringify(value)
+    } catch {
+        return String(value)
+    }
+}
```
[the remark this answers is not in the document]
Comment (dprevost-LMI) [reply src/util/stringUtil.ts:12]: Fixed

File: test/util/stringUtil.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        test('should return string representation for other types', () => {
+            expect(toJsonString(123)).toBe('123')
+            expect(toJsonString(true)).toBe('true')
+            expect(toJsonString(null)).toBe('null')
+        })
+
+        test('should handle undefined correctly', () => {
+            expect(toJsonString(undefined)).toBeUndefined()
+        })
```
[the remark this answers is not in the document]
Comment (dprevost-LMI) [reply test/util/stringUtil.test.ts:47]: Fixed

File: src/util/stringUtil.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dprevost-LMI) [reply src/util/stringUtil.ts:12]: Done
